src/navigator/NavigationService.ts

```diff
@@ -226,12 +226,20 @@ interface NavigateHomeOptions {
   params?: StackParamList[Screens.DrawerNavigator]
 }
 
+/***
+ * Navigates to the home screen resetting the navigation stack by default
+ * If called from a modal make sure to pass fromModal: true. Otherwise it will cause a null pointer dereference and subsequent app crash
+ */
 export function navigateHome(options?: NavigateHomeOptions) {
   const { params } = options ?? {}
-  navigationRef.current?.reset({
-    index: 0,
-    routes: [{ name: Screens.DrawerNavigator, params }],
-  })
+  if (params?.fromModal) {
+    navigate(Screens.WalletHome)
```

Comment:
I think one issue is if at some point we call `navigateHome` from the onboarding flow from a modal, the stack won't be reset.

I don't yet have a great alternative proposal though.

But what about doing something similar to this when `fromModal` is set?

https://github.com/valora-inc/wallet/blob/5c0a7a5889a44a67a87c3df9117a7f8ec783a4fb/src/account/Settings.tsx#L296-L301

Not ideal I know but will reset the stack.

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/4171/commits/99472a5afd2daeb274e2489ff2411236601894fa!